bpf/go_str.h

```diff
@@ -22,6 +22,12 @@ static __inline int read_go_str(char *name, void *base_ptr, u8 offset, void *fie
         return 0;
     }
 
+    // put in a null terminator if we are not at max_size
+    if (size < max_size) {
+        char null = 0;
+        bpf_probe_read((char *)field + size, 1, &null);
```

Comment:
Interesting... Is eBPF complaining if you just try `field[size+1] = 0`?

Reply:
I didn't t try it, but I can before I merge this. I thought with reserve/commit we had to use this approach since they can't prove we are allowed to access the memory. I'll try...

Reply:
Yeah that worked, I should've tried that first :).